graylog2-server/src/main/java/org/graylog2/rest/resources/streams/destinations/filters/StreamDestinationFiltersResource.java

```diff
@@ -0,0 +1,187 @@
+    private final StreamService streamService;
+
+    @Inject
+    public StreamDestinationFiltersResource(StreamDestinationFilterService filterService, StreamService streamService) {
+        this.filterService = filterService;
+        this.streamService = streamService;
+    }
+
+    @GET
+    @Path("/filters")
+    @ApiOperation("Get available filter rules for stream")
+    public PaginatedResponse<StreamDestinationFilterRuleDTO> getPaginatedFiltersForStream(
+            @ApiParam(name = "streamId", required = true) @PathParam("streamId") @NotBlank String streamId,
+            @ApiParam(name = "pagination parameters") @BeanParam PaginationParameters paginationParams
+    ) {
+        checkPermission(RestPermissions.STREAMS_EDIT, streamId);
+        checkStream(streamId);
+
+        final var paginatedList = filterService.findPaginatedForStream(
+                streamId,
+                paginationParams.getQuery(),
+                Sorts.ascending(StreamDestinationFilterRuleDTO.FIELD_TITLE),
+                paginationParams.getPerPage(),
+                paginationParams.getPage(),
+                dtoId -> isPermitted(RestPermissions.STREAM_DESTINATION_FILTERS_READ, dtoId)
+        );
+
+        return PaginatedResponse.create("elements", paginatedList, paginationParams.getQuery());
+    }
+
+    @GET
+    @Path("/target/{targetId}/filters")
+    @ApiOperation("Get available filter rules for stream and target")
+    public PaginatedResponse<StreamDestinationFilterRuleDTO> getPaginatedFiltersForStreamAndTarget(
+            @ApiParam(name = "streamId", required = true) @PathParam("streamId") @NotBlank String streamId,
+            @ApiParam(name = "targetId", required = true) @PathParam("targetId") @NotBlank String targetId,
+            @ApiParam(name = "pagination parameters") @BeanParam PaginationParameters paginationParams
+    ) {
+        checkPermission(RestPermissions.STREAMS_EDIT, streamId);
+        checkStream(streamId);
+
+        final var paginatedList = filterService.findPaginatedForStreamAndTarget(
+                streamId,
+                targetId,
+                paginationParams.getQuery(),
+                Sorts.ascending(StreamDestinationFilterRuleDTO.FIELD_TITLE),
+                paginationParams.getPerPage(),
+                paginationParams.getPage(),
+                dtoId -> isPermitted(RestPermissions.STREAM_DESTINATION_FILTERS_READ, dtoId)
+        );
+
+        return PaginatedResponse.create("elements", paginatedList, paginationParams.getQuery());
+    }
+
+    @GET
+    @Path("/filters/{filterId}")
+    @ApiOperation("Get filter rule for given ID")
+    public Response getFilter(@ApiParam(name = "streamId", required = true) @PathParam("streamId") @NotBlank String streamId,
+                              @ApiParam(name = "filterId", required = true) @PathParam("filterId") @NotBlank String filterId) {
+        checkPermission(RestPermissions.STREAMS_EDIT, streamId);
```

Comment:
its possible to get a Filter Rule for a different streamId

Reply:
Good point. We should scope the `find*` methods to the stream as well. :+1: